sdk/python/feast/infra/offline_stores/offline_store.py

```diff
@@ -20,18 +20,42 @@
 
 from feast.data_source import DataSource
 from feast.feature_view import FeatureView
+from feast.on_demand_feature_view import OnDemandFeatureView
 from feast.registry import Registry
 from feast.repo_config import RepoConfig
 
 
 class RetrievalJob(ABC):
     """RetrievalJob is used to manage the execution of a historical feature retrieval"""
 
+    @property
     @abstractmethod
+    def full_feature_names(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def on_demand_feature_views(self) -> Optional[List[OnDemandFeatureView]]:
+        pass
```

Comment:
Why are we changing the API for the base class here and adding these fields? Are they necessary?

Reply:
NVM, I see why they're needed now. While I think this is okay for a first pass I want to see if we can get away from adding these additional fields on to the job. I think they are pretty inelegant for a user to have to implement themselves.